src/erlfmt.erl

```diff
@@ -423,6 +429,16 @@ try_location(_, Node) when is_tuple(Node) -> erlfmt_scan:get_anno(location, Node
 try_location(_, [Node | _]) when is_tuple(Node) -> erlfmt_scan:get_anno(location, Node);
 try_location(_, _) -> 0.
 
+write_formatted(FileName, Formatted, check) ->
+    Original = read_file(FileName, fun(File) ->
+        erlfmt_scan:read_all(File)
+    end),
```

Comment:
We can use `file:read_file/1` here

```suggestion
    {ok, Original} = file:read_file(FileName),
```

Reply:
I started with that, but it doesn't work for unicode

Reply:
Fixed, thanks again, great catch.